tests/simulation/physics/sources/test_dipole_magnetic.py

```
@@ -0,0 +1,187 @@
+"""Physics simulation tests: magnetic dipole source.
+
+  3a. Magnetic dipole radiates positive power in free space.
+  3b. Electric and magnetic dipoles have comparable total power and
+      consistent radiation patterns (E↔H duality).
+
+Domain: 60×60×60 cells, PML on all faces.
```

Ruff (0.15.9)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)